### DISCUSSION TITLE
Why do tuples start from Item1

### DISCUSSION THREAD

**Discussion by zms9110750** [dotnet/csharplang#8539] [Q&A] [answered]
When we were learning arrays, we learned that computer counting starts from 0. We subconsciously assume that all parameters start from 0.
so , way tuples from Item1?

**Comment by HaloFour** [comment]
That would be a [runtime](https://github.com/dotnet/runtime) question as the `ValueTuple<...>` family of types is provided by the BCL and expected to be used across any language of the .NET ecosystem.  Not all languages have a history of arrays or elements starting at 0, and tuples aren't related to arrays anyway.

**Comment by jnm2** [comment]
One way of thinking that I would offer is that the `0` in `array[0]` is an _offset_ (a distance, a difference in position), whereas the `1` in `Item1` is an _ordinal_ ("which item? the first item").

**Comment by koszeggy** [comment, marked as answer]
> We subconsciously assume that all parameters start from 0.

Those are not parameters or indexed members but fields with explicit names.

> When we were learning arrays, we learned that computer counting starts from 0.

When you access the fields by the [`ITuple[int]`](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.ituple.item) indexer the indexing starts from 0 and it's also documented:

> The index of the specified `Tuple` element. `index` can range from 0 for Item1 of the `Tuple` to one less than the number of elements in the `Tuple`.
